common/lib/common-definitions/api-report/common-definitions.api.md

```diff
@@ -248,19 +254,25 @@ export interface ITelemetryPerformanceEvent extends ITelemetryGenericEvent {
 // @public
 export interface ITelemetryProperties {
     // (undocumented)
-    [index: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
+    [index: string]: OptionallyTaggedTelemetryProperty<TelemetryEventPropertyType>;
 }
 
+// @public (undocumented)
+export type OptionallyTaggedTelemetryProperty<T> = T | ITaggedTelemetryPropertyType<T>;
+
 // @public
 export type ReplaceIEventThisPlaceHolder<L extends any[], TThis> = L extends any[] ? {
     [K in keyof L]: L[K] extends IEventThisPlaceHolder ? TThis : L[K];
 } : L;
 
+// @public
+export type TelemetryBaseEventPropertyType = string | number | boolean | undefined;
+
 // @public
 export type TelemetryEventCategory = "generic" | "error" | "performance";
 
 // @public
-export type TelemetryEventPropertyType = string | number | boolean | undefined;
+export type TelemetryEventPropertyType = string | number | boolean | undefined | (string | number | boolean)[];
```

Comment:
While we're making this changes*, I'd like to also add support for flat objects:  `{ [key: string]: string | number | boolean | null | undefined; }`.  This will allow us to better support the "details" pattern where we put event-specific props that probably don't apply elsewhere inside a single "details" field as JSON.  This way, when logging code can put together a flat object, and then it will get serialized at the last minute.

See https://github.com/microsoft/FluidFramework/pull/11863/files#r984087045

Reply:
And I think `null` should be allowed in the arrays as well

Reply:
Would be good to write a one-pager summarizing the new types we want to support internally and why, and share at Office Hours.